src/util/loanSearchUtils.js

```diff
@@ -155,6 +155,41 @@ export function sortRegions(regions) {
 	return sorted;
 }
 
+/**
+ * Map isoCode to country name
+ *
+ * @param {string} isoCode
+ * @param {array} region
+ * @returns string
+ */
+export function isoToCountryName(isoCode, countryList = []) {
+	const isoMatch = countryList.find(country => country.isoCode === isoCode);
+	console.log(isoMatch);
+	return isoMatch?.name ?? null;
+}
+
+/**
+ * Map isoCode List to region keyed object with country name array
+ *
+ * @param {*} isoCodes
+ * @param {*} regions
+ * @returns
+ */
+export function mapIsoCodesToCountryNames(isoCodes, regions) {
+	const mappedIsos = regions?.reduce((regionObject, region, regionIndex) => {
+		console.log(regionObject);
+		console.log(region);
+		console.log(regionIndex);
+		const countryNames = isoCodes.filter(iso => {
+			return isoToCountryName(iso, region?.countries) ?? false;
+		});
+		console.log(countryNames);
+		return { ...regionObject, [region.region]: countryNames };
+	}, {});
+	console.log(mappedIsos);
```

Comment:
Clearly, these console logs will be removed but this method does return the same data structure that is currently passed into the update location mechanism!